### PR TITLE
feat(search): expose innerQueries response field when performing a `getLogs`

### LINKED ISSUE #583: feat(logs): add inner_queries response field
**Summary**

Please add the new `inner_queries` response field in `1/logs` response object.

```json
    "inner_queries": [
        {
            "index_name": "indexName",
            "query_id": "313453231",
            "offset": 0,
            "user_token": "user1"
        }
    ]
```

For more information about the object please see the documentation:
https://www.algolia.com/doc/api-reference/api-methods/get-logs/#method-response-inner_queries

### PR DESCRIPTION
**Pull request by aseure**
Close #583

### CODE REVIEW

File: src/main/scala/algolia/responses/Logs.scala [hunk 1]
```diff
@@ -39,7 +39,15 @@ case class Log(
     processing_time_ms: String,
     index: Option[String],
     query_nb_hits: Option[String],
-    exhaustive: Option[Boolean]
+    exhaustive: Option[Boolean],
+    inner_queries: Seq[InnerQuery]
+)
+
+case class InnerQuery(
+    indexName: String,
```
Comment (Ant-hem) [inline src/main/scala/algolia/responses/Logs.scala:47]: It should be `index_name`.

File: src/main/scala/algolia/responses/Logs.scala [hunk 2]
```diff
@@ -39,7 +39,15 @@ case class Log(
...
+    query_id: String,
+    offset: Int,
```
Comment (nunomaduro) [inline src/main/scala/algolia/responses/Logs.scala:49]: `offset` is optional.

File: src/main/scala/algolia/responses/Logs.scala [hunk 3]
```diff
@@ -39,7 +39,15 @@ case class Log(
...
     index: Option[String],
     query_nb_hits: Option[String],
-    exhaustive: Option[Boolean]
+    exhaustive: Option[Boolean],
+    inner_queries: Seq[InnerQuery]
+)
+
+case class InnerQuery(
+    indexName: String,
+    query_id: String,
```
Comment (nunomaduro) [inline src/main/scala/algolia/responses/Logs.scala:48]: `query_id` is optional.

File: src/main/scala/algolia/responses/Logs.scala (hunk 1 above)
Comment (aseure) [reply src/main/scala/algolia/responses/Logs.scala]: Thanks for noticing! Fixed.

File: src/main/scala/algolia/responses/Logs.scala (hunk 2 above)
Comment (aseure) [reply src/main/scala/algolia/responses/Logs.scala]: Thanks for noticing! Fixed.

File: src/main/scala/algolia/responses/Logs.scala (hunk 3 above)
Comment (aseure) [reply src/main/scala/algolia/responses/Logs.scala]: Thanks for noticing! Fixed.

File: src/main/scala/algolia/responses/Logs.scala [hunk 4]
```diff
@@ -39,7 +39,15 @@ case class Log(
...
-    exhaustive: Option[Boolean]
+    exhaustive: Option[Boolean],
+    inner_queries: Seq[InnerQuery]
+)
+
+case class InnerQuery(
+    index_name: String,
+    query_id: Option[String],
+    offset: Option[Int],
+    user_token: String
```
Comment (Ant-hem) [inline src/main/scala/algolia/responses/Logs.scala:50]: `user_token` is optional as well.

File: src/main/scala/algolia/responses/Logs.scala [hunk 5]
```diff
@@ -39,7 +39,15 @@ case class Log(
     processing_time_ms: String,
     index: Option[String],
     query_nb_hits: Option[String],
-    exhaustive: Option[Boolean]
+    exhaustive: Option[Boolean],
+    inner_queries: Seq[InnerQuery]
```
Comment (nunomaduro) [inline src/main/scala/algolia/responses/Logs.scala:43]: Not sure about this language, but this is also optional: `Option[Seq[InnerQuery]]`.

**Comment by aseure** [comment]
CircleCI have been hanging for days on this PR, even after different restarts. Tests are clearly passing in every Scala version on my machine so I'll merge this one to close the feature implementation. But it's worth investigating next time it happens. Probably waiting for the v2 would be a better solution.

**Comment by aseure** [comment]
After a full day trying to investigate what may be wrong with the flakiness on CircleCI, I'm leaving things here at the moment. I still think we will better off pushing for a new major version with a new testing suite. The current one has a lot of dependencies between each tests and is rather hard to change with affecting all other tests of the same section. Implementation of our CTS in the next version will be helping in that regard.

I'm now merging this PR as I can confirm that I have no issues running the full test suite on my machine with all Scala versions with support.

**Pull request merged**